### PR TITLE
Route to serve Maps API key

### PR DESCRIPTION
**Pull request by carpenterd777**
Added a route to serve the API key for the Google Maps, the idea being that the app will query our server during run-time for the key before trying to displaying the map element on screen.

This still has basically no security so its not fit for production but its a start.

### CODE REVIEW

File: src/utils/status_code.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum StatusCode {
+	OK = 200,
+}
+
```
Comment (elenirotsides) [inline src/utils/status_code.ts:4]: I like the idea of this file a lot, great idea!

File: src/server.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import { example } from "./rest/example";
```
Comment (elenirotsides) [inline src/server.ts:1]:
```suggestion
```

**Review by elenirotsides — COMMENTED** [review]
Is this related to a certain issue, or just a start on the maps api route?

**Comment by aej11a** [comment]
I don't really understand the concept here - the approach should either be:

1. Keep the API key on the frontend and let the FE use it directly

2. The server uses the API key to make whatever requests are needed and returns the **data** to the FE, not returning the api key itself. 

Since this api key is for rendering, not data fetching, approach 1 is probably more appropriate. There's not really a reason for an API endpoint to serve an API key, because at that point it's even less secure than just keeping it directly in the frontend - since an API endpoint allows a third party to get our API key programmatically.

I think you might be overthinking 😉 let me know if we need to discuss in discord over a call

**Comment by elenirotsides** [comment]
For point 2 that Andrew mentioned: I had just assumed that's what we would be doing when this is actually done. We'd be creating routes that query the api, and the api would return the data we asked for rather than the key itself. Is that the intention behind the routes? I know there's not much to display yet, so perhaps your idea isn't fully shown here through the code yet.

**Comment by aej11a** [comment]
@elenirotsides this is the API key for displaying the map, not any kind of data fetching, so I think approach 2 doesn't apply here, we just need to keep the API key on the frontend

@carpenterd777 see `Restrict your API keys to be used by only the IP addresses, referrer URLs, and mobile apps that need them` here https://support.google.com/googleapi/answer/6310037 - if we do that, it'll be safe to keep the API key just in the frontend app without a serverside component

**Comment by elenirotsides** [comment]
I thought about this a little more and I agree with Andrew - I think it'll be better to call the api from the frontend since it doesn't really need the node server to serve the data. Remind me to not comment on technical decisions before I've had my morning coffee XD

**Comment by carpenterd777** [comment]
> There's not really a reason for an API endpoint to serve an API key, because at that point it's even less secure than just keeping it directly in the frontend - since an API endpoint allows a third party to get our API key programmatically.

Yeah that makes sense. I was wondering if we could use query params as a password or something - but figuring out how to do that without straight-up commiting the password sounds like more trouble than its worth, and it would probably be just as safe to just use a `.env` in the FE.

If you're wondering why I was so hesitant to do the approach above, it was because of this excerpt from the [Security](https://reactnative.dev/docs/security) section of the React Native docs - 

> Never store sensitive API keys in your app code. Anything included in your code could be accessed in plain text by anyone inspecting the app bundle. Tools like react-native-dotenv and react-native-config are great for adding environment-specific variables like API endpoints, but they should not be confused with server-side environment variables, which can often contain secrets and API keys.
>
> If you must have an API key or a secret to access some resource from your app, the most secure way to handle this would be to build an orchestration layer between your app and the resource.

I'll look into restricting the API keys through GCP and I'll close this PR without merging it.

**Pull request closed without merging**